### ISSUE TITLE
[no-undef-components] false positive for "setup" components

### ISSUE THREAD

**Issue by Ky6uk** [vuejs/eslint-plugin-vue#1975]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.23.1
- **eslint-plugin-vue version:** 9.5.0
- **Node version:** 18.4.0
- **Operating System:** Mac OS

**Please show your full configuration:**

```json5
{
    "parserOptions": {
        "parser": "@typescript-eslint/parser",
        "extraFileExtensions": [".vue"]
    },

    "extends": ["plugin:vue/base"],

    "rules": {
        "vue/no-undef-components": "error"
    }
}
```

**What did you do?**	

```vue
<template>
  <AppHeader />
</template>

<script setup lang="ts">
  import AppHeader from '@/components/AppHeader.vue';
</script>
```

**What did you expect to happen?**

No error produced.

**What actually happened?**

```
/private/tmp/eslint-vue-no-undef-components-bug/App.vue
  2:3  error  The '<AppHeader>' component has been used, but 'AppHeader' only refers to a type  vue/no-undef-components

✖ 1 problem (1 error, 0 warnings)
```

**Repository to reproduce this issue**

https://github.com/Ky6uk/eslint-vue-no-undef-components-bug

**Comment by FloEdelmann** [comment]
CC @Amorites, follow-up to #1967

**Comment by ota-meshi** [comment]
I will fix it.
